### PR TITLE
Implement user registration and login on backend

### LINKED ISSUE #15: Implement user registration and login on backend/frontend
Backend is not a server that render any html files. This is service model. Think about backend like about API service.
It should provide possibility to register/login for users.

User login data should be stored in separate database (mongo, sql, etc). For each user we should have such information:
```
User:
 - id      # uuid
 - login
 - pasword hash
 - display name
```

This info could be extended by any other info. For example: `last_login_time`

When user register or login we should check node in memory and create if it is not exists. We should **not** store any sensitive info in knowledge base. I think best would be to store only `user.id`.
Example construction:
```
..user_instance 
   <- user;
   => nrel_user_id: [user.id];;
```

Let develop our frontend by this [free template](https://material-ui.com/store/items/devias-kit/).

- [x] Implement user register on backend
- [x] #22
- [ ] #21
- [x] Cover with tests
- [x] Document API

### PR DESCRIPTION
**Pull request by Flombik**
Resolves #15

### CODE REVIEW

File: web/backend/service/app.py [hunk 1]
```diff
@@ -1,34 +1,82 @@
...
+    # from .sc_memory.helpers.user import create_user_in_memory, check_user_in_memory
+
+    @app.route("/im_teapot")
+    def test_memory():
+        # create_user_in_memory(1)
+        # check_user_in_memory(1)
+
+        return jsonify({"message": "I'm a teapot"}), 418
+
+    return app
```
Comment (deniskoronchik) [inline web/backend/service/app.py:82]: Can we split this function to classes?

File: web/backend/service/app.py [hunk 2]
```diff
@@ -1,34 +1,82 @@
...
+    app.register_blueprint(user.router, url_prefix="/users")
+    app.register_blueprint(swagger.router)
+
+    @app.route("/")
+    def hello_world():
+        return redirect("/docs")
+
+    # from .sc_memory.helpers.user import create_user_in_memory, check_user_in_memory
+
+    @app.route("/im_teapot")
```
Comment (deniskoronchik) [inline web/backend/service/app.py:75]: ?

File: web/backend/service/sc_memory/helpers/user.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+ArcMemberConstPosPerm = ArcType(
+    const=TypeConst.CONST,
+    arc_pos=TypeArcPos.POS,
+    arc_perm=TypeArcPerm.PERM,
+    is_member=True,
+)
+
+
+def create_user_in_memory(user_id: int):
+    tr = TransactionNamesRead(memory.driver)
```
Comment (deniskoronchik) [inline web/backend/service/sc_memory/helpers/user.py:34]: I think we can remove this code for now. We will write that later with better API. It still now work correctly anyway

File: web/backend/service/sc_memory/helpers/user.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    user_instance = tr.create_node(NodeConst, alias="user_instance")
+    _ = tr.create_edge(class_user, user_instance, ArcMemberConstPosPerm)
+    link = tr.create_link_with_content(LinkConst, content=user_id, is_url=False)
+    edge = tr.create_edge(user_instance, link, ArcMemberConstPosPerm)
+    _ = tr.create_edge(nrel_user_id, edge, ArcMemberConstPosPerm)
+    result = tr.run()
+
+
+def check_user_in_memory(user_id: int) -> bool:
+    tr = TransactionNamesRead(memory.driver)
```
Comment (deniskoronchik) [inline web/backend/service/sc_memory/helpers/user.py:71]: Same as previous.

**Review by deniskoronchik — CHANGES_REQUESTED** [review]
Some changes

File: web/backend/service/app.py (hunk 1 above)
Comment (Flombik) [reply web/backend/service/app.py:82]: You mean something like wrappers for each object type? I have an idea how to wrap all this stuff to the one 'App' class now, but I not really understand why we need here more than one class.

File: web/backend/service/app.py (hunk 2 above)
Comment (Flombik) [reply web/backend/service/app.py:75]: Sorry, it's kind of a joke from me.

File: web/backend/service/app.py [hunk 5]
```diff
@@ -1,34 +1,82 @@
...
+    app.config.from_mapping(
+        SECRET_KEY="dev",
+        PROPAGATE_EXCEPTIONS=True,
+        # SQLAlchemy Setup
+        SQLALCHEMY_DATABASE_URI=os.path.join(app.instance_path, "db.sqlite"),
+        SQLALCHEMY_TRACK_MODIFICATIONS=False,
+        # JWT Setup
+        JWT_SECRET_KEY="dev",
+        JWT_ACCESS_TOKEN_EXPIRES=timedelta(days=7),
+        JWT_ERROR_MESSAGE_KEY="message",
```
[the remark this answers is not in the document]
Comment (Flombik) [reply web/backend/service/app.py:25]: Is this comment about 'JWT ERROR MESSAGE KEY' setting? I think in this case it's ok bacause in all code I use `message` key as hardcoded value. But we can setup kinda global setting for message key and then use it here.

File: web/backend/service/app.py (hunk 1 above)
Comment (deniskoronchik) [reply web/backend/service/app.py]: It can be on class. I mean to use some  OOP

**Review by deniskoronchik — APPROVED** [review]

**Pull request merged**